docs/content/guides/dagster/pythonic-config.mdx

```diff
@@ -0,0 +1,246 @@
+    ...
+
+result = materialize(
+    [scoreboard],
+    run_config=RunConfig(
+        ops={
+            "scoreboard": MyDataStructuresConfig(
+                user_names=["Alice", "Bob"],
+                user_scores={"Alice": 10, "Bob": 20},
+            )
+        }
+    ),
+)
+```
+
+### Nested schemas
+
+Schemas can be nested in one another, or in basic Python data structures.
+
+Here, we define a schema which contains a mapping of user names to complex user data objects.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_nested_schema_config endbefore=end_nested_schema_config dedent=4
+from dagster import asset, materialize
+from dagster._config.structured_config import Config
+from dagster._core.definitions.run_config import RunConfig
+from typing import Dict
+
+class UserData(Config):
+    age: int
+    email: str
+    profile_picture_url: str
+
+class MyNestedConfig(Config):
+    user_data: Dict[str, UserData]
+
+@asset
+def average_age(config: MyNestedConfig):
+    ...
+
+result = materialize(
+    [average_age],
+    run_config=RunConfig(
+        ops={
+            "average_age": MyNestedConfig(
+                user_data={
+                    "Alice": UserData(age=10, email="alice@gmail.com", profile_picture_url=...),
+                    "Bob": UserData(age=20, email="bob@gmail.com", profile_picture_url=...),
+                }
+            )
+        }
+    ),
+)
+```
+
+### Union types
+
+Union types are supported using Pydantic [discriminated unions](https://docs.pydantic.dev/usage/types/#discriminated-unions-aka-tagged-unions). Each union type must be a subclass of <PyObject object="Config"/>. The `discriminator` argument to <PyObject object="Field"/> specifies the field that will be used to determine which union type to use.
+
+Here, we define a config schema which takes in a `pet` field, which can be either a `Cat` or a `Dog`, as indicated by the `pet_type` field.
+
+```python file=/guides/dagster/pythonic_config/pythonic_config.py startafter=start_union_schema_config endbefore=end_union_schema_config dedent=4
+from dagster import asset, materialize
+from dagster._config.structured_config import Config
+from dagster._core.definitions.run_config import RunConfig
+from pydantic import Field
+from typing import Union
+from typing_extensions import Literal
+
+class Cat(Config):
+    pet_type: Literal["cat"]
+    meows: int
+
+class Dog(Config):
+    pet_type: Literal["dog"]
+    barks: float
+
+class ConfigWithUnion(Config):
+    pet: Union[Cat, Dog] = Field(..., discriminator="pet_type")
+
+@asset
+def pet_stats(config: ConfigWithUnion):
+    if isinstance(config.pet, Cat):
+        return f"Cat meows {config.pet.meows} times"
+    else:
+        return f"Dog barks {config.pet.barks} times"
+
+result = materialize(
+    [pet_stats],
+    run_config=RunConfig(
+        ops={
+            "pet_stats": ConfigWithUnion(
+                pet=Cat(pet_type="cat", meows=10),
```

Comment:
seems weird that you need to specify both `Cat` and `"cat"` but that looks like a pydantic thing?